### DISCUSSION TITLE
MIT 6.031: Software Construction - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#133] [Announcements]
# MIT 6.031: Software Construction - CS自学指南

CS自学指南

https://csdiy.wiki/%E8%BD%AF%E4%BB%B6%E5%B7%A5%E7%A8%8B/6031/

**Comment by TsuiMinghung** [comment]
有办法可以做课程教材notes里附加链接的练习吗，好像要mit的学生才行

**Reply by dengyunsheng250** [reply]
好像可以爬虫

**Reply by FengZiGG** [reply]
哥，爬虫成功了吗？有没有爬的思路给一个，我爬虫新手。thx

**Reply by shizuka1117** [reply]
如果不会爬虫的话有比较笨一点的办法。在notes页面打开开发者工具，在里面直接找到练习部分对应的代码文件片段，copy下来放到一个html基础模板里面，就可以看了。

**Reply by Fontzs** [reply]
你们说的是Java版的还是ts版本的

**Reply by Kingsman007137** [reply]
problem set怎么提交啊，可以爬测试文件吗

**Comment by Dantetang** [comment]
这个课2016spring的course number 是6.005，B站上面有人搬运了机翻的视频
https://www.bilibili.com/video/BV1Tp4y197XX?spm_id_from=333.999.0.0

**Comment by ShaneHeX** [comment]
有人能拿到最新的sp22的课程资料代码吗

**Reply by sbwcwso** [reply]
请问大佬有找到最新的学习资料吗？

**Reply by ShaneHeX** [reply]
没拿到，用 6.005 老 Java 学习了

**Comment by Met4physics** [comment]
这个课的网页为什么这么奇怪，是打开方式不对吗

**Reply by YHHD** [reply]
对，文档目录一致漂浮在页面上

**Reply by Piper-WD** [reply]
1. 可能是单纯的网速问题
2. 也可能是你需要翻一下墙

**Comment by 2018hahazhufeng** [comment]
sp22的ps下载不到啊 求助

**Comment by Fontzs** [comment]
https://raw.githubusercontent.com/6031/eclipse031/main/setup/eclipse-for-6031.setup好像失效了

**Reply by Baokker** [reply]
我也发现了这个，现在的eclipse已经不支持Java 15了

**Reply by Fontzs** [reply]
169又太久没有更新了，所以我找了另外一个学校的软件工程课，感觉还不错

**Reply by Kingsman007137** [reply]
老哥找了什么课

**Comment by dghtucs** [comment]
建了一个课程群，想讨论可入715675868

**Reply by Yuechooo** [reply]
请问是哪里的群呀？

**Comment by usamimeri** [comment]
目前课程网站已经更新到2023版本，使用2021版本会导致很多资源失效。比如安装eclipse的时候提示repo不存在。

最新课程网站地址：https://web.mit.edu/6.031/www/sp23/

**Reply by wins-m** [reply]
请问 Praxis Tutor 的 personal link 依然需要MIT账号吧？
https://praxistutor.mit.edu/register/openid?tutor=6102-sp23

**Reply by usamimeri** [reply]
是的，我查询了非MIT学生如何注册，是有提供对应的说明
链接见：https://ist.mit.edu/guest-accounts
以及：https://ras.mit.edu/education-and-career-resources/faqs/other/get-mit-kerberos-account-non-mit-personnel#:~:text=Complete%20the%20online%20request%20form,directly%20to%20the%20sponsoree%2Fguest.
但点击进去要填写申请表时会跳转回登录页面，不知道是不是我方法问题，应该是申请不了了

**Reply by usamimeri** [reply]
关于无法使用课程的ts资料，我找到一个平替https://www.typescripttutorial.net/
，总体和课程流程相似，可以参考学习

**Reply by chiarua** [reply]
您好，请问您认为这门课的problem set难度如何，是不是只能本地测试呢？

**Reply by goahead656** [reply]
那 Praxis Tutor 插件还可以使用吗？

**Reply by hychen11** [reply]
同问

**Reply by usamimeri** [reply]
插件我没有安装成功，可能是身份问题。关于问题集，质量还是相当不错的，按它提示一步步实现即可，没法对答案可以和GPT4一起code review

**Comment by engineillgel** [comment]
!

**Comment by goahead656** [comment]
这是我再写的作业，正在完成中，欢迎大家交流学习，[链接](https://github.com/goahead656/6.005_mit_software_construction)

**Reply by ChelsyLan** [reply]
作业资料代码从哪获得啊 求助

**Comment by rinevard** [comment]
推荐学习6.102版本的同学也看看mit WebLab，由于6.102的problem sets和project不公开，个人觉得这个课程可以作为不错的实战。

https://csdiy.wiki/Web开发/mitweb/
https://weblab.mit.edu/schedule/

**Reply by celes17** [reply]
weblab确实不错，谢谢你的推荐！

**Comment by itzhan** [comment]
注意，该课程最新版本的名字已经改了，现在叫MIT 6.102

**Comment by ctfy66** [comment]
找个人一起相互code review , sp16的版本。 vx：ctfy2520

**Comment by 91ac0m0** [comment]
2025 spring 的网址，https://web.mit.edu/6.102/www/sp25/，今年可以下载 reading 对应的 code 了。

**Reply by hefangyinxiao** [reply]
可以做作业不

**Reply by zykkkl** [reply]
课程网站里的reading Exercise点击没反应是什么情况啊

**Comment by wangxi887** [comment]
请问有教springboot的课程吗

**Comment by zykkkl** [comment]
我看现在课程用的是typescript了，有学习Java版有优势还是直接学习typescript版本更好？

**Comment by Ronny200** [comment]
吐槽下，很像学这门课程但是因为作业的原因又很纠结，2016版本插件也有问题，cs50和cs61之类的做的多好。。对新手来说这门课程应该很有价值，有没有朋友知道和这么课程差不多的公开课，国内的也行

**Reply by sugargeng** [reply]
请问最后怎么解决的，我也想学这门

**Reply by sbwcwso** [reply]
基本没有，这门课的作业目前只有 2016 java 版本有。

不过这门课的质量确实是高，只看 Reading 其实也能学到很多了。

**Comment by yiyuanli939** [comment]
2025 sp版本的网址在这里https://web.mit.edu/6.102/www/sp25/

**Comment by SandmeyerX** [comment]


**Reply by SandmeyerX** [reply]
> 25sp的作业好像不能获取？

虽然说有一个公开只读仓库供旁听生使用，但貌似只限MIT学生，因为mit.github需要认证才能拉取

**Comment by gaowqspace** [comment]
2026 spring的课程已经开始更新了：https://web.mit.edu/6.102/www/sp26/

**Reply by SandmeyerX** [reply]
有办法获取problem sets吗

**Reply by gaowqspace** [reply]
我没学这个，只是刚好看到就贴上来。我看现在整个课程主要用TypeScript，自己还是想学下Java

**Reply by gaowqspace** [reply]
请问大家有Java课程推荐啊？

**Comment by 77amyfly** [comment]
这门课到处都找不到视频，大家都怎么学的啊

**Comment by ay94wzh** [comment]
看2021的reading material，做2016的problem sets还行
